autoblocks/_impl/api/client.py

```diff
@@ -230,3 +234,89 @@ def get_dataset(
+        req = self._client.get(f"/testing/local/runs/{encode_uri_component(run_id)}/results")
+        req.raise_for_status()
+        resp = req.json()
+        return [AutoblocksTestCaseResultId(id=result["id"]) for result in resp["results"]]
+
+    def get_ci_test_results(self, run_id: str) -> List[AutoblocksTestCaseResultId]:
+        req = self._client.get(f"/testing/ci/runs/{encode_uri_component(run_id)}/results")
+        req.raise_for_status()
+        resp = req.json()
+        return [AutoblocksTestCaseResultId(id=result["id"]) for result in resp["results"]]
+
+    def get_local_test_result(self, test_case_result_id: str) -> AutoblocksTestCaseResult:
+        req = self._client.get(f"/testing/local/results/{encode_uri_component(test_case_result_id)}")
+        req.raise_for_status()
+        resp = req.json()
+        result = resp["testCaseResult"]
+
+        # Convert events to Event objects
+        events = [
+            Event(
+                id=event.get("id"),
+                trace_id=event.get("trace_id"),
+                message=event.get("message"),
+                timestamp=event.get("timestamp"),
+                properties=event.get("properties", {}),
+            )
+            for event in result.get("events", [])
+        ]
+
+        # Convert evaluations to Evaluation objects
+        evaluations = [
+            Evaluation(
+                evaluatorId=eval.get("evaluatorId"),
+                score=eval.get("score"),
+                passed=eval.get("passed"),
+                metadata=eval.get("metadata", {}),
+            )
+            for eval in result.get("evaluations", [])
+        ]
+
+        return AutoblocksTestCaseResult(**{**result, "events": events, "evaluations": evaluations})
+
+    def get_ci_test_result(self, test_case_result_id: str) -> AutoblocksTestCaseResult:
+        req = self._client.get(f"/testing/ci/results/{encode_uri_component(test_case_result_id)}")
+        req.raise_for_status()
+        resp = req.json()
+        result = resp["testCaseResult"]
+
+        # Convert events to Event objects
+        events = [
+            Event(
+                id=event.get("id"),
+                trace_id=event.get("trace_id"),
+                message=event.get("message"),
+                timestamp=event.get("timestamp"),
+                properties=event.get("properties", {}),
+            )
+            for event in result.get("events", [])
+        ]
+
+        # Convert evaluations to Evaluation objects
+        evaluations = [
+            Evaluation(
+                evaluatorId=eval.get("evaluatorId"),
+                score=eval.get("score"),
+                passed=eval.get("passed"),
+                metadata=eval.get("metadata", {}),
+            )
+            for eval in result.get("evaluations", [])
+        ]
+
+        return AutoblocksTestCaseResult(**{**result, "events": events, "evaluations": evaluations})
```

Comment:
I think duration, run id, and dataset item id are going to fail here because the api returns like "durationMs", so we need to map them explicitly. Let's just do that for all the fields

Reply:
Ok, I also updated the tests so the mock json returns the camel case like runId, and the actual result expects the run_id.